src/Types.hs

```diff
@@ -1365,6 +1397,7 @@ showIdeaSpace :: IdeaSpace -> String
 showIdeaSpace SchoolSpace    = "school"
 showIdeaSpace (ClassSpace c) = showSchoolClass c
 
+-- FIXME HasUILabel: there is already an instance for SchoolClass.
```

Comment:
`HasUILabel` is about human-readable, this is about machine-readable, but the distinction is not very clear and it's possible we got it wrong in a place or two.  I"ll look into this.